src/gui/models/procedureModel.cpp

```diff
@@ -461,3 +462,19 @@ bool ProcedureModel::insertRows(int row, int count, const QModelIndex &parent)
 
     return true;
 }
+
+bool ProcedureModel::removeRows(int row, int count, const QModelIndex &parent)
+{
+    if (!procedure_)
+        return false;
+
+    // Get the scope associated to the parent index
+    auto scope = getScope(parent);
+
+    beginRemoveRows(parent, row, row + count - 1);
+    for (auto i = 0; i < count; ++i)
+        scope->get().removeNode(data(index(row + i, 0), Qt::UserRole).value<std::shared_ptr<ProcedureNode>>());
+    endRemoveRows();
+
```

Comment:
Need to emit `dataChanged` otherwise we don't properly update other things.
```suggestion

    emit(dataChanged(QModelIndex(), QModelIndex());
```

Furthermore, need to call `updateControls()` in `ProcedureWidget::procedureDataChanged()` (line 134) since the relationships between data are validated by the function, but the incorrect values are still displayed in the current control widget because we don't update it.